### PR TITLE
Add link to gifting settings in auto renew dialog

### LINKED ISSUE #69927: Gifting: Link to gift settings in auto-renew modal
We could add a link to the gifting settings page in the auto-renew confirm dialog:

Image

Acceptance criteria:

- [ ] Link to general settings page
- [ ] Confirm copy

### PR DESCRIPTION
**Pull request by candy02058912**
#### Proposed Changes

* Add a link in the auto renew dialog to point users to turn on their gifting banner.

auto-renew

#### Testing Instructions

* Have auto renew turned on.
* Go to Upgrade > My Plan > Manage plan > Auto-renew is ON (or /purchases/subscriptions/:site/:subscription_id)
* Click on the link
* Should direct to Settings > General

Fixes #69927

### CODE REVIEW

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx
Comment (lsl) [inline client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx]:
```suggestion
			"{{strong}}Did you know:{{/strong}} You can allow site visitors to help cover the cost of your site's plan by enabling gift subscriptions in {{a}}Settings > General{{/a}}.",
			{
				components: {
					a: <a href={ `/settings/general/${ siteDomain }` } />,
					strong: <strong />,
				},
			}
```

Align copy with settings page + add some bold to the did you know part.

Screenshot 2022-11-18 at 16-09-18 Purchase Settings ‹ Business Simple Test Site — WordPress com

**Review by lsl — APPROVED** [review]

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx
Comment (lsl) [reply client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx]: ^ Not blocking, just a suggestion

Comment (candy02058912) [reply client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx]: Thanks for adjusting it!

**Comment by candy02058912** [comment]
I am thinking that we should merge this **after** gifting is out.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7633281

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Comment by lsl** [comment]
> I am thinking that we should merge this after gifting is out.

You can drop it behind the gifting config flag if you like, that should only be available for on wpcalypso/dev/testing etc.

**Review by lupus2k — APPROVED** [review]
Looks good @candy02058912

**Review by DavidRothstein — CHANGES_REQUESTED** [review]
This message seems to be showing up in a lot of places where it shouldn't (i.e. on sites that don't even have the gift subscription setting available).  For example:
- On a Jetpack site, when disabling auto-renew on the Jetpack plan
- On a WordPress.com site that doesn't have a plan (for example, when disabling auto-renew on a domain or add-on)

The wording of the message also seems wrong in the case of a site that already has the gift subscription setting enabled?

Finally, I guess I'm not sure the overall purpose here.  The wording of the message seems to be encouraging people to turn the gift subscription setting on after disabling auto-renew, but isn't it true that for many sites, we will be **automatically** turning that setting on as soon as they disable auto-renew, whether they choose to do so or not?  Shouldn't any messaging here talk about that?

**Comment by lsl** [comment]
> On a Jetpack site, when disabling auto-renew on the Jetpack plan
> On a WordPress.com site that doesn't have a plan (for example, when disabling auto-renew on a domain or add-on)

Thanks for catching these, we'll need to look at plan features to decide whether to show this or not.

> Finally, I guess I'm not sure the overall purpose here. 

I can't find the context for this change but I'm wondering if it was maybe proposed before the change to the default behavior.

Does this copy sound better? Do we have a support page setup for this feature yet?

> **Did you know:** Turning auto renew off enables [Subscription Gifting](#) for your site. Manage Subscription Gifting for your site in [Settings > General](#).

**Comment by candy02058912** [comment]
> I can't find the context for this change but I'm wondering if it was maybe proposed before the change to the default behavior.

The Slack thread for context: p1667988224596339-slack-CKZHG0QCR

> Finally, I guess I'm not sure the overall purpose here.

@DavidRothstein We were thinking to guide users to set up their gifting banner by clicking on that link would defer them from turning off auto-renew.

> Does this copy sound better? Do we have a support page setup for this feature yet?
>> Did you know: Turning auto renew off enables [Subscription Gifting](https://github.com/Automattic/wp-calypso/pull/70106#) for your site. Manage Subscription Gifting for your site in [Settings > General](https://github.com/Automattic/wp-calypso/pull/70106#).

I like how this copy reflects what might happen if they've never changed the gifting settings before, but it also seems to be strongly encouraging them to turn off auto-renew. I'd like to go with the original copy.

And I really do think we need to have a support page setup for this because of the 60/7 days rule. Left a comment in Slack: p1669195871419979-slack-C04CLBHNUDN

**Comment by candy02058912** [comment]
> This message seems to be showing up in a lot of places where it shouldn't (i.e. on sites that don't even have the gift subscription setting available).

Thanks for catching this! Added a feature check `hasSubscriptionGifting` to prevent this from showing when gifting is not available to the user.

**Comment by DavidRothstein** [comment]
I think the original copy only makes sense on its own in the case where gifting is already off **and** won't be automatically turned on if they submit the form.

In the case where gifting will be automatically turned on, perhaps it could work if there's also another sentence on the next screen?  This is what the next screen looks like now:

gifting-toggle

I could imagine the above changing to something like this (although it reads kind of awkwardly):

>Auto-renewal is now off. And subscription gifting is turned on, so your site visitors can cover the cost of your next plan and domain renewal even if you don't.
>
>Before you go....

Why **are** we automatically turning gifting on when they submit the form, though?  I can't really figure out where that requirement came from.  I know the goal was to turn it on for users who already had auto-renew off (at the time this feature is rolled out) but it's not clear to me when we additionally decided to toggle it on behind the scenes for people who turned auto-renew off later.  If that behavior were removed, this would all become a lot simpler.

It's also worth nothing that there are multiple ways for auto-renew to be turned off for a subscription, not just the auto-renew toggle (for example, the "Cancel subscription" flow does it, etc).

**Comment by candy02058912** [comment]
> Why are we automatically turning gifting on when they submit the form, though?

The current rules are as the following (https://github.com/Automattic/dotcom-forge/issues/1176#issuecomment-1308222345):

1. If the user never touched the gifting option before, then it defaults to the inverse of auto-renew, meaning:
* auto-renew off -> gifting on
* auto-renew on -> gifting off

2. If the user has set the gifting option to either on/off before, we would persist that value no matter the auto-renew status.

The rules evolved to this current state because the default value had to be dynamic. Then after this discussion (p1667911057231549/1667863537.759979-slack-CKZHG0QCR), we also had to support gifting for people with auto-renew on.

> It's also worth nothing that there are multiple ways for auto-renew to be turned off for a subscription, not just the auto-renew toggle (for example, the "Cancel subscription" flow does it, etc).

After the subscription has been canceled, then the user's plans probably won't fall into the plans that support gifting anymore.

>> Auto-renewal is now off. And subscription gifting is turned on, so your site visitors can cover the cost of your next plan and domain renewal even if you don't.

With the current rules, we can't assume subscription gifting is turned on in this case.
Maybe it would make more sense if we moved the "Did you know" part to this screen? At the very least, it provides a chance for users to take a look at their gifting settings.

Update: saw @lsl has a comment here (pdtkmj-Ks-p2#comment-1245), which would be a better place to continue this discussion.

I'm think it's better to hold this PR off for now until we're sure where we're heading.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7633281).

**Pull request closed without merging**
